metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -250,11 +324,18 @@ def get_platform_from_database_id(self, database_id):
         return platform_name
 
     @lru_cache(maxsize=None)
-    def get_datasource_urn_from_id(self, datasource_id):
+    def get_dataset_info(self, dataset_id: int) -> dict:
         dataset_response = self.session.get(
-            f"{self.config.connect_uri}/api/v1/dataset/{datasource_id}"
-        ).json()
-
+            f"{self.config.connect_uri}/api/v1/dataset/{dataset_id}",
+        )
+        if dataset_response.status_code != 200:
+            logger.warning(f"Failed to get dataset info: {dataset_response.text}")
+            dataset_response.raise_for_status()
```

Comment:
if this can raise an exception, which method is responsible for catching it? where do we report the error to the user?

Reply:
Will contain the `construct_dataset_from_dataset_data` inside of a `try/except` block and raise any errors inside of the except block. Let me know if this approach works/follows best practices for ingestions

Reply:
that sounds good